### ISSUE TITLE
Possible enhancement for responds_with matcher?

### ISSUE THREAD

**Issue by vlad-pisanov** [freerange/mocha#578]
Currently, the matcher `responds_with` takes 2 params: an attribute and a value, like:

```ruby
responds_with(:name, 'Joe')
```

How about introducing a variation of this matcher that takes multiple attribute-value pairs? e.g.

```ruby
responds_with(name: 'Joe', age: 21, salary: 40_000)
```

The matcher would succeed if all attributes match. This would give us a compact way of matching complex objects, and the syntax would be symmetrical to the `has_entries` matcher that we currently have for Hashes:

```ruby
has_entries(name: 'Joe', age: 21, salary: 40_000)
```

**Comment by floehopper** [comment]
@vlad-pisanov Thanks for the suggestion. At first glance it sounds sensible to me. I'll take a proper look at it early next week.

**Comment by floehopper** [comment]
@vlad-pisanov I haven't got much time at the moment, but I managed to find a few minutes to draft [this PR](https://github.com/freerange/mocha/pull/579). Is that roughly what you had in mind?

**Comment by vlad-pisanov** [comment]
@floehopper nice, that's exactly it! I like the idea of using `AllOf.new(...)` under the hood, very semantic

**Comment by floehopper** [comment]
Closed by #579. Sorry it's taken so long!

**Comment by floehopper** [comment]
[Released in v2.2.0](https://github.com/freerange/mocha/blob/main/RELEASE.md#220)
